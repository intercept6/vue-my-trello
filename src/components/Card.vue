<template>
<div class="card">
  <button class="close-button" @click="removeCardFromList">x</button>
  <div class="body">{{ body }}</div>
</div>
</template>

<script>
import {createComponent} from '@vue/composition-api';

export default createComponent({
  name: "Card",
  props: {
    body: {
      type: String,
      required: true,
    },
    listIndex: {
      type: Number,
      required: true,
    },
    cardIndex: {
      type: Number,
      required: true,
    },
  },
  setup(props) {
    const {
      cardIndex,
      listIndex,
    } = props;

    const removeCardFromList = () => {
      if (confirm('本当にこのカードを削除しますか？')) {
        this.$store.dispatch('removeCardFromList',
          {
            cardIndex,
            listIndex,
          });
      }
    };
    return {
      removeCardFromList,
    };
  },
});
</script>
