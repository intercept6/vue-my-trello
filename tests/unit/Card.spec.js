import {shallowMount} from '@vue/test-utils';
import Card from '../../src/components/Card';

describe('Card.vue', () => {
  it('カードの作成', () => {
    const body = 'test card text';
    const wrapper = shallowMount(Card, {
      propsData: {
        body,
        listIndex: 1,
        cardIndex: 1,
      },
    });

    expect(wrapper.html().includes(body)).toBe(true);
  });

  test('スナップショットテスト', () => {
    const wrapper = shallowMount(Card, {
      propsData: {
        body: 'test card text',
        listIndex: 1,
        cardIndex: 1,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

});
