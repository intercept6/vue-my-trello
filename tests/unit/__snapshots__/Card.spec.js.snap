// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card.vue スナップショットテスト 1`] = `
<div
  class="card"
>
  <button
    class="close-button"
  >
    x
  </button>
   
  <div
    class="body"
  >
    test card text
  </div>
</div>
`;
